neuralpredictors/data/datasets/movies.py

```diff
@@ -233,3 +234,47 @@ def transformed_mean(self, stats_source=None):
         if self.rename_output:
             x = self._output_point(*x)
         return x
+
+
+class NRandomSubSequenceDataset(Dataset):
+    """
+    Data augmentation for training.
+    Generate a new dataset based on dat2, by random sampling of each training item in dat2 for multiple times.
+    This only works for movie data and each sampling is a subsequence of the full sequence in a dat2 item.
+    Args:
+        dat2: an original dataset
+        newtiers: list, tiers for each item in new dataset
+        newinds: list, indice for each item in new dataset
+        random_start: array, start positions at each dat2 item for random sampling
```

Comment:
why not to generate `random_start` array inside this augmentation?

Reply:
good idea, done.